### DISCUSSION TITLE
Welcome to magento2-cypress-testing-suite Discussions!

### DISCUSSION THREAD

**Discussion by peterjaap** [elgentos/magento2-cypress-testing-suite#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sprankhub** [comment]
Hi @peterjaap and thanks for your work on this!

I am the owner of the German Magento-only agency [CustomGento](https://www.customgento.com/en/). When starting with Magento 2, we also started using MFTF. We did like the possibility to write frontend tests in a Magento-compatible way. However, we found the following issues over time:

* MFTF is slow
* MFTF tests tend to randomly fail in the pipeline and succeed after you re-run the pipeline without any changes
* MFTF tests tend to break during Magento updates
* MFTF tests are not Hyvä-compatible

Since we also started with Hyvä, we are very interested in this project and hope we can implement it for our clients.

**Comment by lfolco** [comment]
Hi!

I'm a freelancer who, like @sprankhub, started using MFTF a while back and I've run into the same issues. It got to the point where I pretty much stopped writing MFTF tests because they were do time-consuming and they failed so sporadically that I couldn't really trust the output.

I discovered Cypress this summer and have been playing with it here and there. I have one client on Hyvä, so curious how it will fit with that.

**Comment by nisrineMach** [comment]
Hello, thank you for the work and effort put here 

I am the QA at [Webvisum](https://webvisum.de/), and we did start using Cypress in Magento 2 few months ago also with in hyvä.
After few months of work we have already a running test setup working for different frontend scenarios , such as :

-  Creating customer account
-  Login and logout (Magento 2)
-  Checking the customers account pages
-  Add/remove product to wishList
-  Add product to cart as guest and Customer
-  Add multiple product to cart
-  Check viewable Header and Footer 
-  Login into Hyva as Guest and Customer
-  Check the mini Cart 
-  Check quantity in the cart and Mini-cart and change it...

It's saving us time without impacting the performance of the projects

We will be happy to import our contribution and merge those tests here to this project and see how it fit.

**Reply by convenient** [reply]
Hello @nisrineMach 

We have a similar setup, except we're currently on codeception.

>Check viewable Header and Footer

For assertions like this we have had good results in integrating https://percy.io/ for visual regression testing. Especially useful when adding it to preexisting websites that way you don't have to manually code assertions for the minutiae of the site but you can approve the initial build as your baseline and then any changes to header/footer would be caught flagged as the tests went through.

That approach may not be everyones cup of tea, but it allowed for good coverage of whats being added / removed from the frontend in terms of visibility without having to do any code to assert on any specifics of the page.
